x-pack/solutions/observability/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -402,25 +403,30 @@ function getErrorCountByParentId(errorDocs: TraceAPIResponse['traceItems']['erro
   }, {});
 }
 
-export const getOrphanTraceItemsCount = (
-  traceDocs: Array<WaterfallTransaction | WaterfallSpan>
-) => {
-  const waterfallItemsIds = new Set(
-    traceDocs.map((doc) =>
-      doc.processor.event === 'span'
-        ? (doc?.span as WaterfallSpan['span']).id
-        : doc?.transaction?.id
-    )
-  );
+export function getOrphanItemsIds(waterfall: IWaterfallSpanOrTransaction[]) {
+  const waterfallItemsIds = new Set(waterfall.map((item) => item.id));
+  return waterfall
+    .filter((item) => item.parentId && !waterfallItemsIds.has(item.parentId))
+    .map((item) => item.id);
+}
 
-  let missingTraceItemsCounter = 0;
-  traceDocs.some((item) => {
-    if (item.parent?.id && !waterfallItemsIds.has(item.parent.id)) {
-      missingTraceItemsCounter++;
+export function reparentOrphanItems(
+  orphanItemsIds: string[],
+  waterfallItems: IWaterfallSpanOrTransaction[],
+  newParentId?: string
+) {
+  const orphanIdsMap = orphanItemsIds.reduce<Record<string, boolean>>(
```

Comment:
I think here instead of doing the reduce we can just use `orphanItemsIds.includes(item.id)` inside the `if` to simplify it and remove this part and if the performance is an issue we can make this a `Map` and set the values, something like:
```js
const orphanIdsMap = new Map()
orphanItemsIds.every((id)=> orphanIdsMap.set(id, true))
....
if (orphanIdsMap.has(item.id)) {
```
wdyt?

Reply:
Nice, thanks!